src/main/java/org/intellij/sdk/codesync/utils/FileUtils.java

```diff
@@ -0,0 +1,137 @@
+package org.intellij.sdk.codesync.utils;
+
+import org.intellij.sdk.codesync.CodeSyncLogger;
+import org.intellij.sdk.codesync.exceptions.FileInfoError;
+import org.intellij.sdk.codesync.exceptions.FileNotFoundError;
+import org.intellij.sdk.codesync.files.IgnoreFile;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Stream;
+
+public class FileUtils
+{
+    //Read file content into the string with - Files.lines(Path path, Charset cs)
+    public static String readLineByLineJava8(String filePath)
+    {
+        StringBuilder contentBuilder = new StringBuilder();
+
+        try (Stream<String> stream = Files.lines( Paths.get(filePath), StandardCharsets.UTF_8))
+        {
+            stream.forEach(s -> contentBuilder.append(s).append("\n"));
+        }
+        catch (IOException e)
+        {
+            e.printStackTrace();
+        }
+
+        return contentBuilder.toString();
+    }
+
+    public static String readFileToString(String filePath) {
+        return readFileToString(new File(filePath));
+    }
+
+    public static String readFileToString(File file) {
+        try {
+            return org.apache.commons.io.FileUtils.readFileToString(file, StandardCharsets.UTF_8);
+        } catch (IOException e) {
+            CodeSyncLogger.logEvent(String.format("Could read file '%s'. Error: %s", file.getPath(), e.getMessage()));
+            return null;
+        }
+    }
+
+    /*
+        List absolute file paths of all the files in a directory.
+
+        This is recursively list all the files containing in the given directory and all its sub-directories.
+         */
+    public static String[] listFiles(String directory) {
```

Comment:
If there is no .syncigore, it syncs all the files, right?

Reply:
Yes